analyzere/requestor.py

```diff
@@ -33,21 +33,28 @@ def handle_api_error(resp, code):
         raise errors.ServerError(message, body, code, json_body)
 
 
-def request(method, path, params=None, data=None, auto_retry=True):
+def request(method, path, params=None, headers=None, data=None,
+            auto_retry=True):
     """
     method - HTTP method. e.g. get, put, post, etc.
     path - Path to resource. e.g. /loss_sets/1234
     params - Parameter to pass in the query string
+    headers - Dictionary of additional HTTP headers
     data - Dictionary of parameters to pass in the request body
     """
     body = None
     if data is not None:
         body = json.dumps(data, cls=utils.DateTimeEncoder)
 
-    headers = {
-        'accept': 'application/json',
-        'content-type': 'application/json',
-    }
+    # append the default set of headers to any user provided headers
```

Comment:
if you want to force the not-raw request version to use json, I think that's fine.  but otherwise I guess you would just do 

```
def whatever_method(override_dict=None):
    override_dict = override_dict or {}
    default_dict = {
        'accept': 'application/json',
        'content-type': 'application/json',
    }
    default_dict.update(override_dict)
    # then pass the default_dict into whatever needs it
```